src/accounts/context/userAccount.tsx

```diff
@@ -127,28 +128,34 @@ export const UserAccountProvider: FC<Props> = React.memo(({children}) => {
     }
   }
 
-  async function handleRenameActiveAccount(accountId, newName) {
-    const isActiveAcct = acct => acct.isActive
-    const activeIndex = userAccounts.findIndex(isActiveAcct)
-    const oldName = userAccounts[activeIndex].name
-
-    try {
-      const resp = await patchAccount({accountId, data: {name: newName}})
-
-      if (resp.status !== 200) {
-        dispatch(notify(accountRenameError(oldName)))
-      } else {
-        dispatch(notify(accountRenameSuccess(oldName, newName)))
+  const handleRenameActiveAccount = useCallback(
+    async (accountId, newName) => {
+      const activeAccount = userAccounts.find(acct => acct.isActive)
+      try {
+        const accountData = await updateUserAccount(accountId, newName)
         event('multiAccount.renameAccount')
-
-        // change the name, and reset the active accts:
-        userAccounts[activeIndex].name = newName
-        setUserAccounts(userAccounts)
+        dispatch(notify(accountRenameSuccess(activeAccount.name, newName)))
+
+        const updatedAccounts = userAccounts.map(acct => {
+          if (acct.id === activeAccount.id) {
+            return {...acct, name: accountData.name}
+          }
+
+          return acct
+        })
+        setUserAccounts(updatedAccounts)
+      } catch (error) {
+        dispatch(notify(accountRenameError(activeAccount.name)))
+        reportErrorThroughHoneyBadger(error, {
+          name: 'UserAccount',
+          context: {
```

Comment:
Pending the tweak to this `reportErrorThroughHoneyBadger` call - adding a more descriptive name, and adding relevant information about the account to context (account name, account id, user name, or id), this looks good to me.